test/lib/babylon/babylon-test-helper.ts

```diff
@@ -45,4 +49,26 @@ export function initTestEngine(loadableScene: Readonly<LoadableScene>) {
     },
     loadableScene,
   }
+}
+
+export class CrdtBuilder {
+  #buffer = new ReadWriteByteBuffer()
+
+  put<T>(componentId: number, entityId: Entity, timestamp: number, serde: SerDe<T>, value: T) {
+    const componentBuffer = new ReadWriteByteBuffer()
+    serde.serialize(value, componentBuffer)
+    PutComponentOperation.write(entityId, componentId, timestamp,  componentBuffer.toBinary(), this.#buffer)
+    return this
+  }
+
+  delete(componentId: number, entityId: Entity, timestamp: number) {
+    DeleteComponent.write(entityId, componentId, timestamp, this.#buffer)
+    return this
+  }
+
+  toBinary() {
+    const ret = this.#buffer.toBinary()
+    this.#buffer = new ReadWriteByteBuffer()
```

Comment:
_Question_
Why reset the buffer as a side-effect? It's unusual for a builder-type object. I don't want to suggest anything out of ignorance though

Reply:
a proper name would be .finish() instead of toBinary